examples/platform/silabs/efr32/rs911x/sl_wifi_if.c

```diff
@@ -0,0 +1,754 @@
+int32_t wfx_rsi_disconnect()
+{
+    return sl_wifi_disconnect(SL_WIFI_CLIENT_INTERFACE);
+}
+
+sl_status_t join_callback_handler(sl_wifi_event_t event, char * result, uint32_t result_length, void * arg)
+{
+    temp_reset = (wfx_wifi_scan_ext_t *) malloc(sizeof(wfx_wifi_scan_ext_t));
+    memset(temp_reset, 0, sizeof(wfx_wifi_scan_ext_t));
+    if (CHECK_IF_EVENT_FAILED(event))
+    {
+        SILABS_LOG("F: Join Event received with %u bytes payload\n", result_length);
+        callback_status = *(sl_status_t *) result;
+        wfx_rsi.join_retries += 1;
+        wfx_rsi.dev_state &= ~(WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED);
+        wfx_retry_interval_handler(is_wifi_disconnection_event, wfx_rsi.join_retries++);
+        if (is_wifi_disconnection_event || wfx_rsi.join_retries <= WFX_RSI_CONFIG_MAX_JOIN)
+        {
+            xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+        }
+        is_wifi_disconnection_event = true;
+        return SL_STATUS_FAIL;
+    }
+    /*
+     * Join was complete - Do the DHCP
+     */
+    SILABS_LOG("%s: join completed.", __func__);
+    SILABS_LOG("%c: Join Event received with %u bytes payload\n", *result, result_length);
+    xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_CONN);
+    wfx_rsi.join_retries = 0;
+    retryInterval        = WLAN_MIN_RETRY_TIMER_MS;
+    is_wifi_disconnection_event = false;
+    callback_status = SL_STATUS_OK;
+    return SL_STATUS_OK;
+}
+
+#if CHIP_DEVICE_CONFIG_ENABLE_SED
+/******************************************************************
+ * @fn   wfx_rsi_power_save()
+ * @brief
+ *       Setting the RS911x in DTIM sleep based mode
+ *
+ * @param[in] None
+ * @return
+ *        None
+ *********************************************************************/
+int32_t wfx_rsi_power_save()
+{
+    int32_t status;
+    status = rsi_bt_power_save_profile(2, 0);
+    if (status != RSI_SUCCESS)
+    {
+        SILABS_LOG("BT Powersave Config Failed, Error Code : 0x%lX", status);
+        return status;
+    }
+
+    sl_wifi_performance_profile_t wifi_profile = { ASSOCIATED_POWER_SAVE };
+    status = sl_wifi_set_performance_profile(&wifi_profile);
+    if (status != RSI_SUCCESS)
+    {
+        SILABS_LOG("Powersave Config Failed, Error Code : 0x%lX", status);
+        return status;
+    }
+    SILABS_LOG("Powersave Config Success");
+    return status;
+}
+#endif /* CHIP_DEVICE_CONFIG_ENABLE_SED */
+
+/*************************************************************************************
+ * @fn  static int32_t wfx_wifi_rsi_init(void)
+ * @brief
+ *      Wifi initialization called from app main
+ * @param[in]  None
+ * @return
+ *        None
+ *****************************************************************************************/
+int32_t wfx_wifi_rsi_init(void)
+{
+  SILABS_LOG("wfx_wifi_rsi_init started");
+  sl_status_t status;
+  status = sl_wifi_init(&config  , default_wifi_event_handler);
+  if(status != SL_STATUS_OK){
```

Comment:
add check in only one place ,either in caller function or here.

Since you are already check in calling function remove check from here